covsirphy/cleaning/pcr_data.py

```diff
@@ -0,0 +1,439 @@
+
+        Args:
+            country(str): country name or ISO3 code
+            province(str or None): province name
+            start_date(str or None): start date, like 22Jan2020
+            end_date(str or None): end date, like 01Feb2020
+
+        Returns:
+            tuple(pandas.DataFrame, bool):
+                pandas.DataFrame:
+                    Index: reset index
+                    Columns:
+                        - Date(pd.TimeStamp): Observation date
+                        - Tests (int): the number of total tests performed
+                        - Confirmed(int): the number of confirmed cases
+                str/bool: kind of complement or False
+        """
+        country_alias = self.ensure_country_name(country)
+        subset_arg_dict = {
+            "country": country, "province": province,
+            "start_date": start_date, "end_date": end_date,
+        }
+        try:
+            return (self.subset(**subset_arg_dict), False)
+        except ValueError:
+            raise SubsetNotFoundError(
+                country=country, country_alias=country_alias, province=province,
+                start_date=start_date, end_date=end_date) from None
+
+    def positive_rate(self, country, province=None, window=3, show_figure=True, filename=None):
+        """
+        Return the PCR rate of a country as a dataframe.
+
+        Args:
+            country(str): country name or ISO3 code
+            window (int): window of moving average, >= 1
+            show_figure (bool): if True, show the records as a line-plot.
+            filename (str): filename of the figure, or None (show figure)
+
+        Returns:
+            pandas.DataFrame
+                Index:
+                    reset index
+                Columns:
+                    - Date (pd.TimeStamp): Observation date
+                    - Tests (int): the number of total tests performed
+                    - Confirmed (int): the number of confirmed cases
+                    - PCR_Rate (float): positive rate (%) of the daily cases over the total daily tests performed
+
+        Notes:
+            If non monotonic records were found for either cases or tests,
+            "(complemented)" will be added to the title of the figure.
+        """
+        window = self.ensure_natural_int(window, name="window")
+        df = self.records(country, province)[0]
+        if df[self.TESTS].max() == 0:
+            print("No tests records found for country " + country) if not province else print("No tests records found for province " + province)
+            df["PCR_Rate"] = 0
+            return df
+        
+        # Check if there are too many missing value, more than half
+        if len(np.where(df[self.TESTS] == 0)[0]) >= len(df[self.TESTS])/2:
+            print("Too many missing tests records for country " + country) if not province else print("Too many missing tests records for province " + province) 
```

Comment:
Please raise ValueError.
(Or, create `ComplementFailedError` in covsirphy/util/error.py)